nsxt/provider.go

```diff
@@ -210,6 +210,15 @@ func Provider() *schema.Provider {
 				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("NSXT_CA", nil),
 			},
+			// This config is set to true when need to create NSXT manager cluster resource with
+			// vms from vsphere provider as host. Skipping version check will allow terraform
+			// to apply the template while NSX host vm is still being configured.
+			"skip_version_check": {
```

Comment:
The version check should not be skipped, but performed once on demand whenever connector is first used.
In addition to version check, NSX connection is also made to NSX client in `configureNsxtClient` to retrieve session token. This should also happen on demand. I suggest to take this piece to a separate PR.

Reply:
Got it, I have removed it from code